src/transformers/utils/bitsandbytes.py

```diff
@@ -132,8 +133,15 @@ def get_key_to_not_convert(model):
     model (`torch.nn.Module`):
         Input model
     """
+    # Check if the model has tied parameters
+    # This will keep the lm_head etc on their original class type
+    has_tied_params = len(find_tied_parameters(model)) > 0
```

Comment:
Should we force this to be the last tied param just in case it doesn't match the last layer later on?

Reply:
I think that this could break the case where we use `SequenceClassification` models since there no tied params there but I am not sure.
Actually this is called before the first call of `model.tie_weights()`. Therefore this dict was empty for eg for bloom models.
I proposed a small update at: efd78bc739a6ed16890904ccf1f0b5e585ccc389  where the last tied param should match the `last_name` variable later for some cases